plugins/audit-log-node/src/types.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  AuthService,
+  HttpAuthService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { ErrorLike } from '@backstage/errors';
+import { JsonValue } from '@backstage/types';
+
+import { Request } from 'express';
+
+export type ActorDetails = {
+  actorId: string | null;
+  ip?: string;
+  hostname?: string;
+  userAgent?: string;
+};
+
+export type AuditRequest = {
+  body: any;
+  url: string;
+  method: string;
+  params?: any;
+  query?: any;
+};
+
+export type AuditResponse = {
+  status: number;
+  body?: any;
+};
+
+export type AuditLogStatus =
+  | {
+      status: 'failed';
+      errors: ErrorLike[];
+    }
+  | { status: 'succeeded' };
+
+/**
+ * Common fields of an audit log. Note: timestamp and pluginId are automatically added at log creation.
+ *
+ * @public
+ */
+export type AuditLogDetails = {
+  actor: ActorDetails;
+  eventName: string;
+  stage: string;
+  request?: AuditRequest;
+  response?: AuditResponse;
+  meta: JsonValue;
+  isAuditLog: true;
+} & AuditLogStatus;
+
+export type AuditLogDetailsOptions = {
+  eventName: string;
+  stage: string;
+  metadata?: JsonValue;
+  response?: AuditResponse;
+  actorId?: string;
+  request?: Request;
+} & ({ status: 'succeeded' } | { status: 'failed'; errors: unknown[] });
```

Comment:
You could reuse the status types from above here.

Reply:
Ah in this case, I use `unknown[]` instead of `ErrorLike[]` so I couldn't reuse it.